src/chunkserver/chunkserver.cpp

```diff
@@ -260,13 +260,21 @@ int ChunkServer::Run(int argc, char** argv) {
     LOG_IF(FATAL, copysetNodeManager_->Init(copysetNodeOptions) != 0)
         << "Failed to initialize CopysetNodeManager.";
 
+    // init scan model
+    ScanManagerOptions scanOpts;
+    InitScanOptions(&conf, &scanOpts);
+    scanOpts.copysetNodeManager = copysetNodeManager_;
```

Comment:
copysetNodeManager_ = &CopysetNodeManager::GetInstance();
Does not need copysetNodeManager as opts

Reply:
> &CopysetNodeManager::GetInstance();

done

Reply:
> copysetNodeManager_ = &CopysetNodeManager::GetInstance();
> Does not need copysetNodeManager as opts

Yes, but if use CopysetNodeManager::GetInstance() in Scanmanager::Init() will have a trouble to mock this static function in unittest. So add the copysetManager in opts is also fine.

Reply:
> copysetNodeManager_ = &CopysetNodeManager::GetInstance();
> Does not need copysetNodeManager as opts

Yes, but if use CopysetNodeManager::GetInstance() in Scanmanager::Init() will have a trouble to mock this static function in unittest. Maybe need add another class to wrap on CopysetNodeManager. So add the copysetManager in opts is also fine.